de.metas.report/de.metas.report.jasper.client/src/main/java/de/metas/report/jasper/client/interceptor/AD_Process.java

```diff
@@ -47,15 +50,27 @@ public AD_Process()
 
 	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE }, ifColumnsChanged = I_AD_Process.COLUMNNAME_Type)
 	@CalloutMethod(columnNames = I_AD_Process.COLUMNNAME_Type)
-	public void setClassnameIfTypeSQL(final I_AD_Process process)
+	public void setClassnameIfTypeJasperReportsSQL(final I_AD_Process process)
 	{
 		final String processType = process.getType();
-		if (X_AD_Process.TYPE_JasperReports.equals(processType))
+		if (X_AD_Process.TYPE_JasperReportsSQL.equals(processType)
+				|| X_AD_Process.TYPE_JasperReportsJSON.equals(processType))
 		{
 			process.setClassname(JasperReportStarter.class.getName());
 		}
 	}
 
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_CHANGE }, ifColumnsChanged = I_AD_Process.COLUMNNAME_Type)
+	public void checkJSONPathIfTypeJasperReportJSON(final I_AD_Process process)
```

Comment:
i suggest to merge setClassnameIfTypeJasperReportsSQL and checkJSONPathIfTypeJasperReportJSON in one method.